src/scripts/client/modules/hw.js

```diff
@@ -0,0 +1,200 @@
+( function( tiy, moment ) {
+  'use strict';
+
+  const HW_NO_CLIENT_TEMPLATE = 'templates/hw.html';
+
+  let $ui = null;
+  let pageData = {};
+
+  tiy.loadModule( {
+    name: 'hw',
+    navIcon: 'fa-download',
+    render: main
+  } );
+
+  function main( data, elem ) {
+    $ui = $( elem );
+    pageData = data;
+
+    // On relevant pages add links
+    if ( pageData.assignment || pageData.assignment_submission ) {
+      addCloneLinksToUi();
+    }
+
+    $( $ui ).on( 'showing', function() {
+      $ui.append( `<i class='newline_hw_working fa fa-spin fa-spinner'></i>` );
+
+      detectNativeClient( function handleNativeClient( resp ) {
+        addMainContent( $ui, resp );
+      } );
+    } );
+  }
+
+
+  function addMainContent( $ui, resp ) {
+    $ui
+      .html( "" )
+      .append( `<h5>Newline HW ${resp.status === "ok" ? 'detected' : 'not found'}</h5>` );
+
+    if ( resp.status === "ok" ) {
+      $ui.append( `Last Heartbeat at ${moment(resp.message_at).fromNow()}` );
+      $ui.append( `<dl>` );
+
+      for ( let i in resp.data ) {
+        if ( resp.data.hasOwnProperty( i ) ) {
+          $ui.append( `<dt>${i}</dt><dd>${resp.data[i]}</dd>` );
+        }
+      }
+
+      $ui.append( `</dl>` );
+    } else {
+      $.get( chrome.extension.getURL( HW_NO_CLIENT_TEMPLATE ) ).then( function( html ) {
+        $ui.append( html );
+      } );
+    }
+  }
+
+
+  function addCloneLinksToUi() {
+    detectNativeClient( function handleNativeClient( resp ) {
+      if ( resp.status === "ok" ) {
+        if ( pageData.assignment_submission ) {
+          updateUIOnSubmissionPage();
+        }
+        if ( pageData.assignment ) {
+          updateUIOnAssignmentPage();
+        }
+      }
+    } );
+  }
+
+
+  function updateUIOnAssignmentPage() {
+    const idFromUrl = uri => Number( uri.substr( uri.lastIndexOf( '/' ) + 1 ) );
```

Comment:
I would hesitate to move to regex, seems like a performance loss.  But I am fine to move off the fat arrow.